azerothcore/azerothcore-wotlk
# Core/BanManager: Rewrite Ban system

##### CHANGES PROPOSED:
* Added ban manager
* Delete ban functions in world class
* Seperate ban account to (ban account, ban ip, ban character)
* Support world announce

##### ISSUES ADDRESSED: No

##### TESTS PERFORMED:
- [x] Test in game
- [x] Test in build MSVS 2019
 
##### HOW TO TEST THE CHANGES:
1. Apply commit and sql update file
2. Try writing the command to ban 
2.1. `.ban account`
2.2. `.ban character`
2.3. `.ban playeraccount`
2.4. `.ban ip`
3. If you set conf option 
3.1. `ShowBanInWorld = 1` everyone will see a notification about a ban
3.2. `ShowBanInWorld = 0` everything should be as before

##### KNOWN ISSUES AND TODO LIST:

- [x] Test in build Travis
- [ ] Test in game

##### Target branch(es): Master
